updates/htwo-update0011.sql

```diff
@@ -0,0 +1,24 @@
+----------------------------
+--Steuer zu steuer_id umziehen
+----------------------------
+
+ALTER TABLE buchung add steuer_id INT(10) NULL;
+ALTER TABLE buchung add CONSTRAINT fk_buchung_steuer FOREIGN KEY (steuer_id) REFERENCES steuer (id) DEFERRABLE;
+
+----------------------------
+--Fehlende Steuersaetze anlegen
+----------------------------
+INSERT INTO steuer  (name, satz, steuerkonto_id,ust_nr_steuer,ust_nr_bemessung) SELECT DISTINCT CONCAT(CASE WHEN stk.kontotyp_id = 1 THEN 'Umsatzsteuer' ELSE 'Vorsteuer' END,b.steuer,'%'), b.steuer,s.steuerkonto_id,s.ust_nr_steuer,s.ust_nr_bemessung FROM buchung b, konto k, steuer s, konto stk, geschaeftsjahr gj WHERE b.steuer is not null and ((k.id = b.habenkonto_id and (k.kontoart_id != 3 and k.kontoart_id != 5)) or (k.id = b.sollkonto_id and (k.kontoart_id != 3 and k.kontoart_id != 5))) and s.steuerkonto_id is not null and s.id = k.steuer_id and stk.id = s.steuerkonto_id and gj.id = b.geschaeftsjahr_id and concat(stk.kontenrahmen_id,stk.kontotyp_id, b.steuer) not in (SELECT DISTINCT concat(stk2.kontenrahmen_id,stk2.kontotyp_id, s2.satz) FROM steuer s2,konto stk2 WHERE stk2.kontotyp_id is not null and (s2.mandant_id = gj.mandant_id or s2.mandant_id is null) and s2.steuerkonto_id = stk2.id);
```

Comment:
Können wir die Migration nicht weglassen? Die hatte doch bei dem anderen PR auch nicht funktioniert. Wenn weiterhin der Steuersatz als Dezimalwert in den Buchungen erhalten bleibt, ist die Erweiterung doch abwärtskompatibel und fehlende Steuer-Objekte müssen nicht angelegt werden, weil bei den Bestandsdaten das Steuerobjekt noch aus dem Soll-/Haben-Konto ermittelt werden kann.

Reply:
Hier migriere ich nicht die Buchungen sondern füge nur die fehlenden Steuern in die Steuer Tabelle ein. Das geht bei alle drei Datenbankenarten (Nur das Update Select zum migrieren der Buchungen geht nicht bei McKoi). Wenn man das nicht macht, würde bei Buchungen die einen Abweichenden Steuersatz gespeichert haben <Keine Steuer> beim bearbeiten angezeigt werden und nach dem speichern wäre die Steuer weg. Bei abgeschlossen Geschäftsjahren passiert hier kein Fehler, bei offenen kann es jedoch zu Fehlern kommen wenn der Nutzer nicht auf das Steuerfeld achtet...